src/tests/test_magic.py

```diff
@@ -2248,3 +2248,46 @@ def test_warn_when_using_snippets_in_non_select_command(
 )
 def test_get_query_type(query, query_type):
     assert get_query_type(query) == query_type
+
+
+@pytest.mark.parametrize(
+    "query, expected",
+    [
+        (
+            """%%sql
+SELECT 1""",
+            1,
+        ),
+        (
+            """%%sql
+SELECT 1 -- comment""",
+            1,
+        ),
+        (
+            """%%sql
+SELECT 1
+-- comment""",
+            1,
+        ),
+        (
+            """%%sql
+SELECT 1; -- comment""",
+            1,
+        ),
+        (
+            """%%sql
+SELECT 1;
+-- comment""",
+            1,
+        ),
+        (
+            """%%sql
+SELECT 1; -- comment
+SELECT 2""",
+            2,
+        ),
+    ],
+)
+def test_query_trailing_comment(ip, query, expected):
```

Comment:
`test_query_comment_after_semicolon` might be more clear

Reply:
[Fixed](https://github.com/ploomber/jupysql/pull/939/files#diff-d327143929530bcece2cae65d59465708cdf87f499fd83b4bd939c332150f713R2332)